test/scripts/__snapshots__/ast-patches.spec.js.snap

```diff
@@ -0,0 +1,12 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Ast patches ChangeWixAuthElevateFunctionType Should apply patch 1`] = `
+"declare module 'wix-auth' {
+    /**
+     * Returns an instance of the specified function for which the site visitor's permissions have been elevated to the highest level required.
+     * 	[Read more](https://www.wix.com/corvid/reference/wix-auth.html#elevate)
+     */
+    function elevate<T extends (...arg: any) => any>(func: T): (...param: Parameters<T>) => ReturnType<T>;
```

Comment:
isn't it the same?
`export declare function elevate<T extends (...arg: any) => any>(func: T): T;`

Reply:
its what i get from the branch of the techwriters, that comes through docworks, so im sure it's the correct one

Reply:
wdym correct one?
I think that those 2 definitions are the same.

Reply:
No its not the same ![Screen Shot 2021-12-21 at 14 57 57 (2)](https://user-images.githubusercontent.com/85441594/146934090-894775ad-e0d6-402a-963c-f28b91ae4df3.png)

Reply:
> did you test it with the real docworks output?

Yes I checked it with the branch that i got from the techwriter that i generated with dockworks (`DOCS_BRANCH=yaron-auth`)

Reply:
> No its not the same
i see that you are returning an anonymous function with the same singature. 
the thing is that i don't understand how the use of this elevate function will look like and what it should do.
can you please provide an example?

Reply:
and can you please describe the autocomplete experience it should provide

Reply:
example: 
```
interface User {id: string, name: string}

function getUser(id: string): User {
  return {id: '', name: ''}
} 

// old impl
getUser(userId) // -> permission restricted
getUser(userId, {suppressAuth: true}) // -> permission unrestricted 

//new impl
const elevatedGetUser = elevate(getUser);

getUser(userId) // -> permission restricted
elevatedGetUser(userId) // -> permission unrestricted
```
we need the elevated function to still have the same autocompletion as the regular one (gets a `userId: string` and to return a `User`)

![Screen Shot 2021-12-22 at 11 05 50 (2)](https://user-images.githubusercontent.com/85441594/147066854-a3921167-5a15-49e7-9b46-88ab90d4cc45.png)

![Screen Shot 2021-12-22 at 11 06 17 (2)](https://user-images.githubusercontent.com/85441594/147066871-2ebeabd7-e394-46d7-a591-f5aff1c3fdfe.png)

Reply:
cool thanks. 
so why not using
`export declare function elevate<T extends (...arg: any) => any>(func: T): T;`
it will do the trick and it's a much simpler definition

Reply:
can you please send a link to the implementation of the elevate function?

Reply:
https://github.com/wix-private/elementory-dynamic-modules/blob/master/wix-auth/src/elevate.ts

Reply:
> so why not using
> export declare function elevate<T extends (...arg: any) => any>(func: T): T; ?
> it will do the trick and it's a much simpler definition

when you do it like that the types aren't correct in the elevate function
![Screen Shot 2021-12-22 at 11 38 16 (2)](https://user-images.githubusercontent.com/85441594/147071409-d2f01f0f-6769-41a3-9f98-1d1286689685.png)